In [1]:
import numpy as np
import time

Random function is used to create a random element and rand function is used to create a array of those elements.

In this example 10000 random values between 0,1 is created and stored in the variable a

In [23]:
a=np.random.rand(100000)
b=np.random.rand(100000)

In [24]:
start_time=time.time()
c=np.dot(a,b)
end_time=time.time()
print(c)
print(f"the time taken is {(end_time-start_time)*1000} ms")

25123.397592469366
the time taken is 4.410028457641602 ms


Now you can see that the vectorization have taken 4.41 sec to complete the task

In [25]:
c=0
start=time.time()
for i in range(10000):
  c+=a[i]*b[i]
end=time.time()
print(c)
print(f"the time taken is {(end-start)*1000} ms")

2524.9594684877193
the time taken is 11.289358139038086 ms


But the non vectorization have taken 11.2 sec which is three time longer than vectorization. so this is the simple concept of vectorization and it is done using .dot function in numpy.